src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -32,27 +21,27 @@ public LRUCache(int cap) {
 
     private void setCapacity(int newCapacity) {
         checkCapacity(newCapacity);
-        for (int i = data.size(); i > newCapacity; i--) {
-            Entry<K, V> evicted = evict();
-            data.remove(evicted.getKey());
-        }
         this.cap = newCapacity;
```

Comment:
**Performance Issue**: Removing elements in a loop based on capacity check can be inefficient for large data sizes. This approach can lead to high CPU usage and slow down the performance due to constant resizing and rehashing of the HashMap. <br> **Fix**: Consider leveraging a linked list alongside the HashMap to efficiently remove the least recently used item without iterating over the map. This can significantly optimize the eviction process. <br> **Code Suggestion**: 
 ```
 -        for (int i = data.size(); i > newCapacity; i--) {
 -            Entry<K, V> evicted = evict();
 -            data.remove(evicted.getKey());
 -        }
 +        // Assume linked list implementation is used for ordering.
 +        while (data.size() > newCapacity) {
 +            Entry<K, V> evicted = evictLast();
 +            data.remove(evicted.getKey());
 +        }
 ```